### ISSUE TITLE
"Value NIL is not of the expected type REAL" error (stumpwm compiled with ccl on FreeBSD11)

### ISSUE THREAD

**Issue by pbgc** [stumpwm/stumpwm#284]
I tried to compile StumpWM on FreeBSD11 with ccl.
Running it I always get (immediately after start and pressing any key or mouse button):

```
Value NIL is not of the expected type REAL.
 (821C19858) : 0 (PRINT-CALL-HISTORY :CONTEXT NIL :PROCESS NIL :ORIGIN NIL :DETAILED-P NIL :COUNT 100 :START-FRAME-NUMBER 0 :STREAM #<STRING-OUTPUT-STREAM  #x302000D2BACD> :PRINT-LEVEL 2 :PRINT-LENGTH 5 :SHOW-INTERNAL-FRAMES NIL :FORMAT :TRADITIONAL) 853
 (821C199C0) : 1 (PRINT-BACKTRACE 100) 149
 (821C199E0) : 2 (BACKTRACE-STRING) 189
 (821C19A38) : 3 (FUNCALL #'#<#<STANDARD-METHOD STUMPWM::HANDLE-TOP-LEVEL-CONDITION (SERIOUS-CONDITION)>> #<TYPE-ERROR #x302000D2BBAD>) 533
 (821C19A68) : 4 (FUNCALL #'#<(:INTERNAL STUMPWM::STUMPWM-INTERNAL-LOOP)> #<TYPE-ERROR #x302000D2BBAD>) 53
 (821C19A88) : 5 (SIGNAL #<TYPE-ERROR #x302000D2BBAD>) 989
 (821C19AE0) : 6 (%ERROR #<TYPE-ERROR #x302000D2BBAD> NIL 4365759348) 117
 (821C19B08) : 7 (FUNCALL #'#<CCL::DEFAULT-REQUIRE-TYPE-RESTARTS> 4365759348 NIL REAL) 1093
 (821C19BA0) : 8 (<-2 NIL 268245972) 2301
 (821C19BD8) : 9 (FUNCALL #'#<#<STANDARD-METHOD IO-LOOP (STUMPWM::XLIB-IO-LOOP)>> #<STUMPWM::XLIB-IO-LOOP #x302000D0328D> :DESCRIPTION "StumpWM") 2173
 (821C19C88) : 10 (STUMPWM-INTERNAL-LOOP) 837
 (821C19CE8) : 11 (STUMPWM-INTERNAL ":0") 3629
 (821C19D90) : 12 (STUMPWM ":0") 413
 (821C19DB0) : 13 (FUNCALL #'#<(:INTERNAL CCL::%SAVE-APPLICATION-INTERNAL)>) 85
 (821C19DC8) : 14 (FUNCALL #'#<(:INTERNAL CCL::MAKE-MCL-LISTENER-PROCESS)>) 661
 (821C19E60) : 15 (RUN-PROCESS-INITIAL-FORM #<PROCESS toplevel(2) [Active] #x302000701D5D> (#<CCL:COMPILED-LEXICAL-CLOSURE # #x30200070188F>)) 709
 (821C19EF0) : 16 (FUNCALL #'#<(:INTERNAL (CCL::%PROCESS-PRESET-INTERNAL (CCL:PROCESS)))> #<PROCESS toplevel(2) [Active] #x302000701D5D> (#<CCL:COMPILED-LEXICAL-CLOSURE # #x30200070188F>)) 573
 (821C19F98) : 17 (FUNCALL #'#<(:INTERNAL CCL::THREAD-MAKE-STARTUP-FUNCTION)>) 277
```

On the same computer/system If I compile with sbcl everything works without any problem.

**Comment by PuercoPop** [comment]
@pbgc I can't reproduce on Linux with CCL ("Version 1.12-dev-r16660  (LinuxX8664)").

What CCL version are you on?

Also if you could [use Xephr](https://github.com/stumpwm/stumpwm/wiki/Debugging-reproducible-crashing-bugs-or-when-crash-StumpWM-is-likely-to-Crash) to use slime's debugger to find out where the error occurs. To do so, in the backtrace go the stack you are interested in and press v. I'm interested in the 8'th (`(< -2 nil 268245972)`).

I'm thinking the problem is related to a timer-time being nil

**Comment by pbgc** [comment]
@PuercoPop I'm using CCL 1.11-r16635 (FreebsdX8664) on FreeBSD 11.0 RELEASE-p1

I will try to do what you asked ... but can't promise to be fast :) ... I'm a long time programmer but a Lisp newbie ...

**Comment by PuercoPop** [comment]
@pbgc That is the stable release of CCL so you should be ok. Let me know if the wiki is not sufficiently clear or if you need any additional help.

**Comment by dbjergaard** [comment]
@PuercoPop I agree this is the next step for debugging. I fleshed out your wiki page to make it a little more "step by step".  The one thing missing from your prescription was how to load stumpwm into slime before launching stumpwm.  I've opted for `(asdf:load-system :stumpwm)` but I know this will pick up whatever is in asdf's path (ie a previously loaded quicklisp version).  Do you know how to make explicit which path asdf should load/compile the sources from?

**Comment by pbgc** [comment]
@PuercoPop yes , it's the stable release. I compile StumpWM without any problem but after when I try to use it the error is always popping up! (I can launch a terminal and emacs etc and use them .. but with the error on always on top).
Like I said If I compile with sbcl I don't have the same problem.

**Comment by pbgc** [comment]
@PuercoPop I tried to follow the wiki ...
When I do (asdf:load-system :stumpwm) I get: 
Component :STUMPWM not found
   [Condition of type ASDF/FIND-SYSTEM:MISSING-COMPONENT]

I used slime-cd to change to the stumpwm cloned repository directory and checked it with (ccl::current-directory-name) ...

I'm sorry if this is basic stuff ...

**Comment by pbgc** [comment]
@PuercoPop @dbjergaard 
After reading what @dbjergaard  wrote ... I consulted quicklisp documentation ...
I cloned the source into ~/quicklisp/local-projects and used (ql:quickload "stumpwm") and it worked ...
Let me now see if I can debug this thing :)

**Comment by dbjergaard** [comment]
@pbgc bonus points if you can edit the wiki to include/edit the wiki to make the prescription bullet proof :)

**Comment by pbgc** [comment]
@PuercoPop I think I need some help...
I followed all the steps .. and I'm able to launch stumpvm on the Xephyr buffer without any problem.
I trigger the crash as usual (just press a key or mouse button) and the error appears but I get no debugger/stack trace ... just a message on slime repl .. (I attach a screenshot).

I tried to force a crash inside slime on another dummy code and the debugger triggered ... so I think it's not my setup ...
![2016-10-24-175520_1366x768_scrot](https://cloud.githubusercontent.com/assets/180825/19655408/ae307138-9a13-11e6-9feb-e67bc44a0e6c.png)

**Comment by dbjergaard** [comment]
@pbgc looks like you got it going! The point of doing it from slime in emacs is so that you can copy paste the whole stack trace without taking a screenshot.  Can you do that now?

**Comment by PuercoPop** [comment]
> Do you know how to make explicit which path asdf should load/compile the sources from?
@dbjergaard AFAIK ASDF doesn't allow you to specify load this system from this file, only to set the paths. You can however ask ASDF where it is loading the system from. Quicklisp provides the convenience function`(ql:where-is-system <foo>)`. which just calls `asdf:system-source-directory`

ie
```lisp
(defun where-is-system (name)
  "Return the pathname to the source directory of ASDF system with the
given NAME, or NIL if no system by that name can be found known."
  (let ((system (asdf:find-system name nil)))
    (when system
      (asdf:system-source-directory system))))
```

**Comment by pbgc** [comment]
@PuercoPop @dbjergaard 

this ?

```

Caught 'The value NIL is not of the expected type REAL.' at the top level. Please report this.
 (824619290) : 0 (PRINT-CALL-HISTORY :CONTEXT NIL :PROCESS NIL :ORIGIN NIL :DETAILED-P NIL :COUNT 100 :START-FRAME-NUMBER 0 :STREAM #<STRING-OUTPUT-STREAM  #x302001DF01FD> :PRINT-LEVEL 2 :PRINT-LENGTH 5 :SHOW-INTERNAL-FRAMES NIL :FORMAT :TRADITIONAL) 853
 (8246193F8) : 1 (PRINT-BACKTRACE 100) 149
 (824619418) : 2 (BACKTRACE-STRING) 189
 (824619470) : 3 (FUNCALL #'#<#<STANDARD-METHOD STUMPWM::HANDLE-TOP-LEVEL-CONDITION (SERIOUS-CONDITION)>> #<TYPE-ERROR #x302001DF080D>) 533
 (8246194A0) : 4 (FUNCALL #'#<(:INTERNAL STUMPWM::STUMPWM-INTERNAL-LOOP)> #<TYPE-ERROR #x302001DF080D>) 53
 (8246194C0) : 5 (SIGNAL #<TYPE-ERROR #x302001DF080D>) 989
 (824619518) : 6 (%ERROR #<TYPE-ERROR #x302001DF080D> NIL 4371264187) 117
 (824619540) : 7 (FUNCALL #'#<CCL::DEFAULT-REQUIRE-TYPE-RESTARTS> 4371264187 NIL REAL) 1093
 (8246195D8) : 8 (<-2 NIL 76966480) 2301
 (824619610) : 9 (FUNCALL #'#<#<STANDARD-METHOD IO-LOOP (STUMPWM::XLIB-IO-LOOP)>> #<STUMPWM::XLIB-IO-LOOP #x302001DF6F9D> :DESCRIPTION "StumpWM") 2173
 (8246196C0) : 10 (STUMPWM-INTERNAL-LOOP) 837
 (824619720) : 11 (STUMPWM-INTERNAL ":1") 3629
 (8246197C8) : 12 (STUMPWM ":1") 413
 (8246197E8) : 13 (CALL-CHECK-REGS STUMPWM ":1") 221
 (824619820) : 14 (CHEAP-EVAL (STUMPWM ":1")) 101
 (824619858) : 15 (EVAL-REGION "(stumpwm:stumpwm \":1\")
") 365
 (8246198E8) : 16 (FUNCALL #'#<(:INTERNAL SWANK-REPL::REPL-EVAL)>) 61
 (824619908) : 17 (TRACK-PACKAGE #<CCL:COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK-REPL::REPL-EVAL) #x302001D2113F>) 349
 (824619968) : 18 (CALL-WITH-RETRY-RESTART "Retry SLIME REPL evaluation request." #<CCL:COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK-REPL::REPL-EVAL) #x302001D211BF>) 533
 (8246199C0) : 19 (CALL-WITH-BUFFER-SYNTAX NIL #<CCL:COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK-REPL::REPL-EVAL) #x302001D211FF>) 221
 (824619A00) : 20 (REPL-EVAL "(stumpwm:stumpwm \":1\")
") 197
 (824619A18) : 21 (CALL-CHECK-REGS SWANK-REPL:LISTENER-EVAL "(stumpwm:stumpwm \":1\")
") 221
 (824619A50) : 22 (CHEAP-EVAL (SWANK-REPL:LISTENER-EVAL "(stumpwm:stumpwm \":1\")
")) 101
 (824619A88) : 23 (EVAL-FOR-EMACS (SWANK-REPL:LISTENER-EVAL "(stumpwm:stumpwm \":1\")
") "COMMON-LISP-USER" 6) 1381
 (824619BA8) : 24 (PROCESS-REQUESTS NIL) 293
 (824619BF8) : 25 (FUNCALL #'#<(:INTERNAL SWANK::HANDLE-REQUESTS)>) 437
 (824619C60) : 26 (FUNCALL #'#<(:INTERNAL SWANK::HANDLE-REQUESTS)>) 405
 (824619C98) : 27 (FUNCALL #'#<SWANK/BACKEND:CALL-WITH-DEBUGGER-HOOK> #<Compiled-function SWANK:SWANK-DEBUGGER-HOOK #x302000F65C5F> #<CCL:COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK::HANDLE-REQUESTS) #x3020011326FF>) 181
 (824619D20) : 28 (CALL-WITH-BINDINGS ((*STANDARD-INPUT* . #) (*STANDARD-OUTPUT* . #) (*TRACE-OUTPUT* . #) (*ERROR-OUTPUT* . #) (*DEBUG-IO* . #<# input #, output # #x3020011286AD>) ...) #<CCL:COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK::HANDLE-REQUESTS) #x3020011326BF>) 949
 (824619D78) : 29 (HANDLE-REQUESTS #<MULTITHREADED-CONNECTION #x302000FD5B5D> NIL) 1045
 (824619E60) : 30 (RUN-PROCESS-INITIAL-FORM #<PROCESS repl-thread(13) [Active] #x30200111F93D> (#<CCL:COMPILED-LEXICAL-CLOSURE # #x3020011321DF>)) 709
 (824619EF0) : 31 (FUNCALL #'#<(:INTERNAL (CCL::%PROCESS-PRESET-INTERNAL (CCL:PROCESS)))> #<PROCESS repl-thread(13) [Active] #x30200111F93D> (#<CCL:COMPILED-LEXICAL-CLOSURE # #x3020011321DF>)) 573
 (824619F98) : 32 (FUNCALL #'#<(:INTERNAL CCL::THREAD-MAKE-STARTUP-FUNCTION)>) 277

```

**Comment by PuercoPop** [comment]
@pbgc I think the the code that is preventing you to see the SLIME debugger is [this](https://github.com/stumpwm/stumpwm/blob/master/stumpwm.lisp#L162). Try setting *top-level-action* to `:break` ie `(setf stumpwm:*top-level-error-action* :break)` and you should get the debugger. Then move to the 8'th frame and press V and Slime should take you to file the error is occurring on

**Comment by pbgc** [comment]
@PuercoPop  Ok ... it worked!
Now in the debugger stack trace I think it was 5 that interested you (the same than the 8 before).
Using v, opens "l0-numbers.lisp" on line 256

```
(defun <-2 (x y)
  (number-case x
    (fixnum (number-case y
              (fixnum (< (the fixnum x) (the fixnum y)))
              (double-float (eq -1 (fixnum-dfloat-compare x y)))
              (short-float (eq -1 (fixnum-sfloat-compare x y)))
              (bignum (bignum-plusp y))
......
```

The stacktrace is:


```
The value NIL is not of the expected type REAL.
   [Condition of type TYPE-ERROR]

Restarts:
 0: [USE-VALUE] Use a new value of type REAL instead of NIL.
 1: [RESTART-IOLOOP] Restart at I/O loop (StumpWM)
 2: [QUIT-IOLOOP] Quit I/O loop (StumpWM)
 3: [NEW-IO-LOOP] Recreate I/O loop
 4: [QUIT-STUMPWM] Quit Stumpwm
 5: [RETRY] Retry SLIME REPL evaluation request.
 --more--

Backtrace:
  0: (#<STANDARD-METHOD STUMPWM::HANDLE-TOP-LEVEL-CONDITION (SERIOUS-CONDITION)> #<TYPE-ERROR #x302001ED1E6D>)
  1: ((:INTERNAL STUMPWM::STUMPWM-INTERNAL-LOOP) #<TYPE-ERROR #x302001ED1E6D>)
  2: (SIGNAL #<TYPE-ERROR #x302001ED1E6D>)
  3: (CCL::%ERROR #<TYPE-ERROR #x302001ED1E6D> NIL 4371264187)
  4: (CCL::DEFAULT-REQUIRE-TYPE-RESTARTS 4371264187 NIL REAL)
  5: (CCL::<-2 NIL 141726939)
      Locals:
        X = NIL
        Y = 141726939
        #:G229838 = 3
  6: (#<STANDARD-METHOD STUMPWM:IO-LOOP (STUMPWM::XLIB-IO-LOOP)> #<STUMPWM::XLIB-IO-LOOP #x302001DB975D> :DESCRIPTION "StumpWM")
  7: (STUMPWM::STUMPWM-INTERNAL-LOOP)
  8: (STUMPWM::STUMPWM-INTERNAL ":1")
  9: (STUMPWM:STUMPWM ":1")
      Locals:
        DISPLAY-STR = ":1"
 10: (CCL::CALL-CHECK-REGS STUMPWM:STUMPWM ":1")
 11: (CCL::CHEAP-EVAL (STUMPWM:STUMPWM ":1"))
 12: (SWANK::EVAL-REGION "(stumpwm:stumpwm \":1\")\n")
 13: ((:INTERNAL SWANK-REPL::REPL-EVAL))
 14: (SWANK-REPL::TRACK-PACKAGE #<COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK-REPL::REPL-EVAL) #x302001D671FF>)
 15: (SWANK::CALL-WITH-RETRY-RESTART "Retry SLIME REPL evaluation request." #<COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK-REPL::REPL-EVAL) #x302001D6727F>)
 16: (SWANK::CALL-WITH-BUFFER-SYNTAX NIL #<COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK-REPL::REPL-EVAL) #x302001D672BF>)
 17: (SWANK-REPL::REPL-EVAL "(stumpwm:stumpwm \":1\")\n")
 18: (CCL::CALL-CHECK-REGS SWANK-REPL:LISTENER-EVAL "(stumpwm:stumpwm \":1\")\n")
 19: (CCL::CHEAP-EVAL (SWANK-REPL:LISTENER-EVAL "(stumpwm:stumpwm \":1\")\n"))
 20: (SWANK:EVAL-FOR-EMACS (SWANK-REPL:LISTENER-EVAL "(stumpwm:stumpwm \":1\")\n") "COMMON-LISP-USER" 7)
 21: (SWANK::PROCESS-REQUESTS NIL)
```

**Comment by PuercoPop** [comment]
@pbgc ok, so I was mistaken, it was `<` with argument `-2` but function `<-2` which is a special case CCL handles of < for an arity of  two. Could you check where V takes you in these two frames?

```
  6: (#<STANDARD-METHOD STUMPWM:IO-LOOP (STUMPWM::XLIB-IO-LOOP)> #<STUMPWM::XLIB-IO-LOOP #x302001DB975D> :DESCRIPTION "StumpWM")
  7: (STUMPWM::STUMPWM-INTERNAL-LOOP)
```

**Comment by pbgc** [comment]
@PuercoPop 
```
  6: (#<STANDARD-METHOD STUMPWM:IO-LOOP (STUMPWM::XLIB-IO-LOOP)> #<STUMPWM::XLIB-IO-LOOP #x302001DB975D> :DESCRIPTION "StumpWM")
      Locals:
        INFO = #<STUMPWM::XLIB-IO-LOOP #x302001DB975D>
        DESCRIPTION = "StumpWM"
        *CURRENT-IO-LOOP* = #<STUMPWM::XLIB-IO-LOOP #x302001DB975D>
        #:QUIT-IOLOOP = #<RESTART STUMPWM::QUIT-IOLOOP #x82353BABD>
        #:G25069 = (#<RESTART STUMPWM::QUIT-IOLOOP #x82353BABD>)
        CCL::%RESTARTS% = ((#<RESTART STUMPWM::QUIT-IOLOOP #x82353BABD>) (#<RESTART :NEW-IO-LOOP #x82353BC2D>) (#<RESTART STUMPWM::QUIT-STUMPWM #x82353BDDD>) (#<RESTART SWANK::RETRY #x82353C11D>) ..))
        #:CHANNEL-TIMEOUT = #<Compiled-function (:INTERNAL STUMPWM::CHANNEL-TIMEOUT (STUMPWM:IO-LOOP (STUMPWM::XLIB-IO-LOOP))) (Non-Global)  #x302001BFCDDF>
        #:RESTART-IOLOOP = #<RESTART STUMPWM::RESTART-IOLOOP #x82353B99D>
        #:G25088 = (#<RESTART STUMPWM::RESTART-IOLOOP #x82353B99D>)
        CCL::%RESTARTS% = ((#<RESTART STUMPWM::RESTART-IOLOOP #x82353B99D>) (#<RESTART STUMPWM::QUIT-IOLOOP #x82353BABD>) (#<RESTART :NEW-IO-LOOP #x82353BC2D>) (#<RESTART STUMPWM::QUIT-STUMPWM #x82353BDDD>) ..)))
        NOW = 141726939
        #:G25123 = (#<STUMPWM::STUMPWM-TIMER-CHANNEL #x302001DB94ED>)
        CHANNEL = #<STUMPWM::STUMPWM-TIMER-CHANNEL #x302001DB94ED>
```
takes me to "ioloop.lisp" at line 400

```
...
 (when (< (channel-timeout channel) now)
....
```

AND

```
  7: (STUMPWM::STUMPWM-INTERNAL-LOOP)
      Locals:
        #:NEW-IO-LOOP = #<RESTART :NEW-IO-LOOP #x82353BC2D>
        #:G25700 = (#<RESTART :NEW-IO-LOOP #x82353BC2D>)
        CCL::%RESTARTS% = ((#<RESTART :NEW-IO-LOOP #x82353BC2D>) (#<RESTART STUMPWM::QUIT-STUMPWM #x82353BDDD>) (#<RESTART SWANK::RETRY #x82353C11D>) (#<RESTART ABORT #x82353C38D>) ..)))
        IO = #<STUMPWM::XLIB-IO-LOOP #x302001DB975D>
        #:G25710 = (T #<Compiled-function (:INTERNAL STUMPWM::STUMPWM-INTERNAL-LOOP) (Non-Global)  #x302001C2DFDF>)
        CCL::%HANDLERS% = ((T #<Compiled-function (:INTERNAL STUMPWM::STUMPWM-INTERNAL-LOOP) (Non-Global)  #x302001C2DFDF>) (T #<COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK:EVAL-FOR-EMACS) #x302001D672FF>) ..)))
```
takes me to: "stumpwm.lisp" at line 227

```
...
 (io-loop io :description "StumpWM")))))))
...

```

**Comment by PuercoPop** [comment]
@pbgc Perfect, So the error seems to be that a [channel-timeout](https://github.com/stumpwm/stumpwm/blob/master/ioloop.lisp#L400) is nil.

**Comment by PuercoPop** [comment]
@pbgc Sorry to bother you again, Would you mind posting the Slots of the Channel? To do so go to ` CHANNEL = #<STUMPWM::STUMPWM-TIMER-CHANNEL #x302001DB94ED>` and between the `#< >` press I to open the inspector on that instance and paste that text please.

**Comment by pbgc** [comment]
@PuercoPop No bother at all! I'm sorry I cannot help more .. but feel completely free to ask what you need.
I did what you said but go:

```
Unbound variable: STUMPWM::STUMPWM-TIMER-CHANNEL
   [Condition of type UNBOUND-VARIABLE]

Restarts:
 0: [RESTART-IOLOOP] Restart at I/O loop (StumpWM)
 1: [QUIT-IOLOOP] Quit I/O loop (StumpWM)
 2: [NEW-IO-LOOP] Recreate I/O loop
 3: [QUIT-STUMPWM] Quit Stumpwm
 4: [RETRY] Retry SLIME REPL evaluation request.
 5: [*ABORT] Return to SLIME's top level.
 --more--

Backtrace:
  0: (#<Anonymous Function #x302001F5E35F>)
  1: (CCL::CHEAP-EVAL (LET ((STUMPWM:INFO '#<STUMPWM::XLIB-IO-LOOP #x302001E6825D>) ..))))
  2: ((:INTERNAL SWANK/CCL::CALL/FRAME) 4371264194 NIL)
  3: (CCL::%MAP-CALL-FRAMES-INTERNAL #<COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK/CCL::CALL/FRAME) #x302001E9DF6F> NIL 4371264143 1152921504606846969 6 NIL)
  4: (MAP-CALL-FRAMES #<COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK/CCL::CALL/FRAME) #x302001E9DF6F> :CONTEXT NIL :PROCESS NIL :ORIGIN 4371264143 :COUNT 1152921504606846969 :START-FRAME-NUMBER 6 :TEST NIL)
  5: (SWANK/CCL::CALL/FRAME 6 #<COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK/BACKEND:EVAL-IN-FRAME) #x302001E9DFBF>)
  6: ((:INTERNAL SWANK:INSPECT-IN-FRAME))
  7: (SWANK::CALL-WITH-RETRY-RESTART "Retry SLIME inspection request." #<COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK:INSPECT-IN-FRAME) #x302001E9E30F>)
  8: (SWANK::CALL-WITH-BUFFER-SYNTAX NIL #<COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK:INSPECT-IN-FRAME) #x302001E9B78F>)
  9: (CCL::CALL-CHECK-REGS SWANK:INSPECT-IN-FRAME "STUMPWM::STUMPWM-TIMER-CHANNEL" 6)
 10: (CCL::CHEAP-EVAL (SWANK:INSPECT-IN-FRAME "STUMPWM::STUMPWM-TIMER-CHANNEL" 6))
 11: (SWANK:EVAL-FOR-EMACS (SWANK:INSPECT-IN-FRAME "STUMPWM::STUMPWM-TIMER-CHANNEL" 6) "COMMON-LISP-USER" 23)
 12: (SWANK::SLDB-LOOP 1)
 13: (SWANK/BACKEND:CALL-WITH-DEBUGGING-ENVIRONMENT #<Compiled-function (:INTERNAL SWANK::DEBUG-IN-EMACS) (Non-Global)  #x302000F9964F>)
 14: (SWANK::DEBUG-IN-EMACS #<TYPE-ERROR #x302001E8A2DD>)
 15: (SWANK:INVOKE-SLIME-DEBUGGER #<TYPE-ERROR #x302001E8A2DD>)
 16: (SWANK/BACKEND:CALL-WITH-DEBUGGER-HOOK #<Compiled-function SWANK:SWANK-DEBUGGER-HOOK #x302000F9B2FF> #<COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK:SWANK-DEBUGGER-HOOK) #x302001E8A1EF>)
 17: (SWANK:SWANK-DEBUGGER-HOOK #<TYPE-ERROR #x302001E8A2DD> #<Compiled-function SWANK:SWANK-DEBUGGER-HOOK #x302000F9B2FF>)
 18: (INVOKE-DEBUGGER #<TYPE-ERROR #x302001E8A2DD>)
 19: (#<STANDARD-METHOD STUMPWM::HANDLE-TOP-LEVEL-CONDITION (SERIOUS-CONDITION)> #<TYPE-ERROR #x302001E8A2DD>)
 20: ((:INTERNAL STUMPWM::STUMPWM-INTERNAL-LOOP) #<TYPE-ERROR #x302001E8A2DD>)
 21: (SIGNAL #<TYPE-ERROR #x302001E8A2DD>)
 22: (CCL::%ERROR #<TYPE-ERROR #x302001E8A2DD> NIL 4371264187)
 23: (CCL::DEFAULT-REQUIRE-TYPE-RESTARTS 4371264187 NIL REAL)
 24: (CCL::<-2 NIL 282267161)
 25: (#<STANDARD-METHOD STUMPWM:IO-LOOP (STUMPWM::XLIB-IO-LOOP)> #<STUMPWM::XLIB-IO-LOOP #x302001E6825D> :DESCRIPTION "StumpWM")
 26: (STUMPWM::STUMPWM-INTERNAL-LOOP)
 27: (STUMPWM::STUMPWM-INTERNAL ":1")
 28: (STUMPWM:STUMPWM ":1")
 29: (CCL::CALL-CHECK-REGS STUMPWM:STUMPWM ":1")
 30: (CCL::CHEAP-EVAL (STUMPWM:STUMPWM ":1"))
 31: (SWANK::EVAL-REGION "(stumpwm:stumpwm \":1\")\n")
 32: ((:INTERNAL SWANK-REPL::REPL-EVAL))
 33: (SWANK-REPL::TRACK-PACKAGE #<COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK-REPL::REPL-EVAL) #x302001DD455F>)
 34: (SWANK::CALL-WITH-RETRY-RESTART "Retry SLIME REPL evaluation request." #<COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK-REPL::REPL-EVAL) #x302001DD45DF>)
 35: (SWANK::CALL-WITH-BUFFER-SYNTAX NIL #<COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK-REPL::REPL-EVAL) #x302001DD461F>)
 36: (SWANK-REPL::REPL-EVAL "(stumpwm:stumpwm \":1\")\n")
 37: (CCL::CALL-CHECK-REGS SWANK-REPL:LISTENER-EVAL "(stumpwm:stumpwm \":1\")\n")
 38: (CCL::CHEAP-EVAL (SWANK-REPL:LISTENER-EVAL "(stumpwm:stumpwm \":1\")\n"))
 39: (SWANK:EVAL-FOR-EMACS (SWANK-REPL:LISTENER-EVAL "(stumpwm:stumpwm \":1\")\n") "COMMON-LISP-USER" 16)
 40: (SWANK::PROCESS-REQUESTS NIL)
 41: ((:INTERNAL SWANK::HANDLE-REQUESTS))
 42: ((:INTERNAL SWANK::HANDLE-REQUESTS))
 43: (SWANK/BACKEND:CALL-WITH-DEBUGGER-HOOK #<Compiled-function SWANK:SWANK-DEBUGGER-HOOK #x302000F9B2FF> #<COMPILED-LEXICAL-CLOSURE (:INTERNAL SWANK::HANDLE-REQUESTS) #x30200114E66F>)
 44: (SWANK::CALL-WITH-BINDINGS ((*STANDARD-INPUT* . #<SWANK/GRAY::SLIME-INPUT-STREAM #x30200114D83D>) (*STANDARD-OUTPUT* . #<SWANK/GRAY::SLIME-OUTPUT-STREAM #x30200114D4CD>) ..))) #<COMPILED-LEXICAL-CLOSU..
 --more--
```

**Comment by jorams** [comment]
This should be fixed by b91f257a5caaaa93244bfa752551c735291894ab.

The problem was caused by the assumption that also caused #272 and #276: that there will always be at least one active timer.

**Comment by pbgc** [comment]
I can confirm that is fixed.
